modules/fedimint-ln-client/src/db.rs

```diff
@@ -80,3 +88,94 @@ impl_db_record!(
 );
 
 impl_db_lookup!(key = MetaOverridesKey, query_prefix = MetaOverridesPrefix);
+
+fn get_v1_migrated_state(
+    bytes: &[u8],
+    module_instance_id: ModuleInstanceId,
+    decoders: &ModuleDecoderRegistry,
+) -> anyhow::Result<Option<DynState>> {
+    let mut cursor = std::io::Cursor::new(bytes);
+    let key = fedimint_core::core::ModuleInstanceId::consensus_decode(&mut cursor, decoders)?;
+    debug_assert_eq!(key, module_instance_id, "Unexpected module instance ID");
+
+    let ln_sm_variant = u16::consensus_decode(&mut cursor, decoders)?;
+
+    // If the state machine is not a receive state machine, return None
+    if ln_sm_variant != 2 {
+        return Ok(None);
+    }
+
+    let _ln_sm_len = u16::consensus_decode(&mut cursor, decoders)?;
+    let operation_id = OperationId::consensus_decode(&mut cursor, decoders)?;
+    let receive_sm_variant = u16::consensus_decode(&mut cursor, decoders)?;
+
+    // If the receive state machine is not a SubmittedOffer variant, return None
+    if receive_sm_variant != 0 {
+        return Ok(None);
+    }
```

Comment:
Normally we don't have to think about this too much, we could also make a deep copy of the LN state machine structure definition tree, but that would be a lot of copied code. So instead we just manually implement the relevant parts of the parsing code that would otherwise be derived.

Maybe we need documentation for our encoding scheme, but documenting each version of a every struct is a waste of time imo. You can read the struct/enum definitions and infer the serialized format if encoding/decoding is derived or you can read the manual impls if not.